File: colockitchenrace/CohouseUser.swift
//
//  CohouseUser.swift
//  colockitchenrace
//
//  Created by Julien Rahier on 03/07/2024.
//

import Foundation

struct CohouseUser: Codable, Equatable, Hashable, Identifiable {
    var id: UUID
    var isAdmin: Bool = false
    var surname: String = ""
    var userId: String?
}

extension CohouseUser {
    static var mock: CohouseUser {
        CohouseUser(
            id: UUID(),
            isAdmin: false,
            surname: "Julien"
        )
    }
}

File: android/app/src/main/java/dev/rahier/colockitchenrace/data/model/CohouseUser.kt
package dev.rahier.colockitchenrace.data.model

import java.util.UUID

data class CohouseUser(
    val id: String = UUID.randomUUID().toString(),
    val isAdmin: Boolean = false,
    val surname: String = "",
    val userId: String? = null,
) {
    val isAssignedToRealUser: Boolean get() = userId != null
}
